crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -293,7 +348,17 @@ def upgrade_controller_nodes
         # and the existing cluster has been cleaned up by deleting most of resources
         return false unless node_api.post_upgrade
 
-        # FIXME: if upgrade went well, continue with next node(s)
+        # Delete router namespaces on non-upgraded node
+        master_node_api.delete_router_namespace
```

Comment:
Would it make sense to implement this as a follow-up PR? This would help move forward for the rest of the code.

FWIW, on how to implement this: with HA, controller nodes can ssh into each other, so we could have a script running from the upgraded controller that loops over the namespaces in the second controller, and do what @rhafer suggests.